### PR TITLE
added timeout flag to multicluster check command

### LINKED ISSUE #11266: Value of `--wait` flag is being used as a timeout by `linkerd multicluster check` command
### What problem are you trying to solve?

All check commands in the codebase support a `--wait` flag. That flag is used as the value of "RetryDeadline" when executing individual checks. For the `linkerd multicluster check` command _only_, the value of the flag is also being used as a timeout when checking if gateway mirrors have endpoints. This is confusing, since the `--wait` duration controls whether or not to retry failed checks in almost all cases, but it also controls how long it takes to run an individual check in just this one case.

### How should the problem be solved?

We could introduce a separate `--timeout` flag that can be used to time out individual checks. That way the `--wait` flag can be used to control retry behavior exclusively, and not affect timeouts.

### Any alternatives you've considered?

We could just pick a sane timeout for checking if gateway mirrors have endpoints, and not allow that duration to be configurable via CLI flags.

### How would users interact with this feature?

_No response_

### Would you like to work on this feature?

maybe

### PR DESCRIPTION
**Pull request by moki1202**
Fixes #11266 

Signed-off-by: Shashank <shashanksingh819@gmail.com>

### CODE REVIEW

**Comment by moki1202** [comment]
I'm not sure about the exact default duration for the timeouts so please let me know and I'll make the changes.

**Review by olix0r — APPROVED** [review]

File: multicluster/cmd/check.go [hunk 1]
```diff
@@ -107,6 +109,7 @@ non-zero exit code.`,
 	}
 	cmd.Flags().StringVarP(&options.output, "output", "o", options.output, "Output format. One of: table, json, short")
 	cmd.Flags().DurationVar(&options.wait, "wait", options.wait, "Maximum allowed time for all tests to pass")
+	cmd.Flags().DurationVar(&options.timeout, "timeout", options.timeout, "timeout duration between running checks")
```
Comment (hawkw) [inline multicluster/cmd/check.go:112]: All the other help text starts with an uppercase letter, so let's uppercase this as well.
```suggestion
	cmd.Flags().DurationVar(&options.timeout, "timeout", options.timeout, "Timeout duration between running checks")
```

Comment (hawkw) [reply multicluster/cmd/check.go:112]: Also, I'm not sure if I love the wording "between running checks", that kind of implies that we wait for this period of time after running a check prior to starting the next one...

**Review by hawkw — CHANGES_REQUESTED** [review]

File: multicluster/cmd/check.go (hunk 1 above)
Comment (adleong) [reply multicluster/cmd/check.go:112]: this is specifically a timeout for calls to the kubernetes API, so we probably want to be clear about that in the flag description.

Comment (moki1202) [reply multicluster/cmd/check.go:112]: @hawkw @adleong I had my suspicions about the description too! I'll push the changes, although I still don't completely understand what this timeout duration exactly mean.

Comment (adleong) [reply multicluster/cmd/check.go:112]: How about: "Timeout for calls to the Kubernetes API"?

Comment (olix0r) [reply multicluster/cmd/check.go:112]:
```suggestion
	cmd.Flags().DurationVar(&options.timeout, "timeout", options.timeout, "Timeout for calls to the Kubernetes API")
```

**Review by hawkw — APPROVED** [review]
looks good to me!

**Review by adleong — APPROVED** [review]

**Pull request merged**